### PR TITLE
WACCMX-FV bug fix:convert wet species to dry before calling cam_thermo_dry_air_update

### LINKED ISSUE #690: Corrections to se_sponge_del4 namelist variables definitions
### Issue Type

Other (please describe below)

### Issue Description

Namelist variable `se_sponge_del4_nu_div_fac` needs to be a real, while `se_sponge_del4_lev` needs to be an integer.   

The corrections to `namelist_definition.xml` are as follows:

```
diff --git a/bld/namelist_files/namelist_definition.xml b/bld/namelist_files/namelist_definition.xml
index d371f975..516f973d 100644
--- a/bld/namelist_files/namelist_definition.xml
+++ b/bld/namelist_files/namelist_definition.xml
@@ -7626,7 +7626,7 @@ If &lt; 0, se_sponge_del4_nu_fac is automatically set based on model top locatio
 Default: Set by build-namelist.
 </entry>
 
-<entry id="se_sponge_del4_nu_div_fac" type="integer" category="se"
+<entry id="se_sponge_del4_nu_div_fac" type="real" category="se"
        group="dyn_se_inparm" valid_values="" >
 Divergence damping hyperviscosity coefficient se_nu_div [m^4/s] for u,v is increased to
 se_nu_p*se_sponge_del4_nu_div_fac following a hyperbolic tangent function
@@ -7640,7 +7640,7 @@ If &lt; 0, se_sponge_del4_nu_div_fac is automatically set based on model top loc
 Default: Set by build-namelist.
 </entry>
 
-<entry id="se_sponge_del4_lev" type="real" category="se"
+<entry id="se_sponge_del4_lev" type="integer" category="se"
        group="dyn_se_inparm" valid_values="" >
 Level index around which increased del4 damping is centered.
```

### Will this change answers?

I Don't Know

### Will you be implementing this yourself?

Yes

### LINKED ISSUE #885: Bug in WACCM-x FV introduced in cam6_3_109
### What happened?

Bug is associated with computing cpairv, rairv and molecular and thermal conductivity coefficients (kmvis, kmcnd).

In cam6_3_108 the calls to update these variables (one in dp_coupling and one in physics_update)

call cam_thermo_update

passed moist major species ('O', 'O2', 'H', 'N2') mixing ratios. (in dp_coupling mixing ratios already wet, in call from physics_types they are converted to wet in the cam_thermo_update call)

In cam6_3_109 the call in dp_coupling used moist major species mixing ratios but in physics_update dry mixing ratios were passed. This had significant impact on FV WACCM-x simulations (very cold summer mesopause temperature and very large gravity wave forcing as reported by @hanli-liu in WACCM meeting 9/7/2023).

Also: note that WACCM-x bugs  for MPAS and SE (in geopotential.F90, dp_coupling) have been fixed in https://github.com/ESCOMP/CAM/pull/849.

### What are the steps to reproduce the bug?

COMPSET FX2000 with FV dycore f09_f09_mg17

### What CAM tag were you using?

cam6_3_108 and cam6_3_109

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

Yes

### Extra info

Bug fix is here:

https://github.com/PeterHjortLauritzen/CAM/blob/clubb_hb/src/dynamics/fv/dp_coupling.F90

We should always use dry mixing ratios to update cpairv, rairv, etc.!

### LINKED ISSUE #893: Add ability to archive derecho baselines
### What is the feature/what would you like to discuss?

Baselines generated on derecho need to be stored via archive_baseline.sh

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Yes

### PR DESCRIPTION
**Pull request by jtruesdal**
This PR corrects a bug introduced in cam6_3_109.  Physics requires constituents mixing ratios to be dry but was using constituent thermodynamic properties based on the wet mixing ratio returned by dynamics.  To fix this bug, the routine called to convert mixing ratios from wet to dry was moved further up in d_p_coupling so that it is called before the thermodynamic properties are calculated for physics. 

An additional fix to close issue #690 was added to correct namelist default types which are used to control the effect of the SE sponge layer. se_sponge_del4_nu_div_fac has been redefined from and integer to a real type as it controls real valued coefficient.  se_sponge_del4_lev was defined as a real but identifies a specific integer model level and has been redefined from a real to integer type.

closes #885
closes #690 
closes #893

### CODE REVIEW

**Comment by PeterHjortLauritzen** [comment]
Fixed https://github.com/ESCOMP/CAM/issues/690

File: src/dynamics/fv/dp_coupling.F90 [hunk 1]
```diff
@@ -571,7 +571,12 @@ subroutine d_p_coupling(grid, phys_state, phys_tend,  pbuf2d, dyn_out)
              end if
           end do
        end do
-
+       !
+       ! Convert dry type constituents from moist to dry mixing ratio
+       ! (note: cam_thermo_dry_air_update assumes dry unless optianal conversion factor provided)
+       !
+       call set_state_pdry(phys_state(lchnk))    ! First get dry pressure to use for this timestepxxx
+       call set_wet_to_dry(phys_state(lchnk))    ! Dynamics had moist, physics wants dry. xxx
```
Comment (cacraigucar) [inline src/dynamics/fv/dp_coupling.F90:579]: What are `xxx` in these comments for?  Do they need to be removed?

**Review by cacraigucar — CHANGES_REQUESTED** [review]

File: src/dynamics/fv/dp_coupling.F90 (hunk 1 above)
Comment (PeterHjortLauritzen) [reply src/dynamics/fv/dp_coupling.F90]: yes. Done

**Comment by cacraigucar** [comment]
@jtruesdal FYI - to get the automated closing to happen, you'll need to remove the word "issue" between "closes" and the issue #

**Review by cacraigucar — APPROVED** [review]

**Comment by jtruesdal** [comment]
Thanks.  Removed Issue.

File: src/dynamics/fv/dp_coupling.F90 [hunk 2]
```diff
@@ -571,7 +571,12 @@ subroutine d_p_coupling(grid, phys_state, phys_tend,  pbuf2d, dyn_out)
              end if
           end do
        end do
-
+       !
+       ! Convert dry type constituents from moist to dry mixing ratio
+       ! (note: cam_thermo_dry_air_update assumes dry unless optianal conversion factor provided)
```
Comment (nusbaume) [inline src/dynamics/fv/dp_coupling.F90:576]: Typo here (`optianal` -> `optional`).

**Review by nusbaume — APPROVED** [review]
Looks good!  Just found one very minor typo.

File: src/dynamics/fv/dp_coupling.F90 (hunk 2 above)
Comment (jtruesdal) [reply src/dynamics/fv/dp_coupling.F90:576]: Corrected.

**Pull request merged**